CHANGELOG.md

```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ## [Unreleased 2.x]
 ### Added
 - Renaming the node role search to warm ([#17573](https://github.com/opensearch-project/OpenSearch/pull/17573))
+- Introduce a new search node role ([#17620](https://github.com/opensearch-project/OpenSearch/pull/17620))
```

Comment:
Lets add a bit more context on what its for - Introduce a new search node role to hold search only shards